Abstract numeric state

kk7ds opened the pull request.

kk7ds committed:
Add state_as_number() helper

This adds state_as_number(), a helper method that tries to interpret
state as a number, for cases we can predict. It's a generalization of
what is copy-and-paste-ed into multiple other places.

kk7ds committed:
Convert statsd, influx, splunk, and graphite to use state_as_number()

Fixes #1205

kk7ds commented on homeassistant/core.py:
```diff
@@ -373,6 +374,36 @@ def from_dict(cls, json_dict):
+        elif self.state is None:
+            return 0
+        elif isinstance(self.state, bool):
+            return int(self.state)
```
Somewhere in here (while trying to be thorough) I forgot that state is always going to be a str.

balloob commented on homeassistant/core.py:
```diff
@@ -373,6 +374,35 @@ def from_dict(cls, json_dict):
         return cls(json_dict['entity_id'], json_dict['state'],
                    json_dict.get('attributes'), last_changed, last_updated)
 
+    def as_number(self):
```
I have tried to keep the core component agnostic. I would suggest we move this to the helpers.state module

kk7ds replied:
Sure, so usage would be:

```
  helpers.state.state_as_number(state)
```

?

balloob replied:
Yep

On Thu, Feb 11, 2016 at 4:25 PM, Dan Smith <notifications@github.com> wrote:

> In homeassistant/core.py
> <https://github.com/balloob/home-assistant/pull/1209#discussion_r52690498>
> :
>
> > @@ -373,6 +374,35 @@ def from_dict(cls, json_dict):
> >          return cls(json_dict['entity_id'], json_dict['state'],
> >                     json_dict.get('attributes'), last_changed, last_updated)
> >
> > +    def as_number(self):
>
> Sure, so usage would be:
>
>   helpers.state.state_as_number(state)
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/1209/files#r52690498>.
>



-- 
PaulusSchoutsen.nl
It's nice to be important but it's more important to be nice.

kk7ds committed:
Add simple statsd tests

These are not very amazing, but at least exercise the code a little
to make sure I didn't break anything. Hopefully they're useful in the
future too.

kk7ds committed:
Add some tests for splunk

This also fixes issue #1214, and I think another bug. The splunk
code will just take the value of state.state and try to serialize
it to json if it can't make it into a number. It did this before
I generalized that code. Since json.dumps() will fail on most anything
complicated, I think the right thing to do is *not* try to do that.

balloob commented:
Looks great :dolphin:
